frontend/packages/ceph-storage-plugin/src/components/modals/volume-snapshot-modal/volume-snapshot-modal.tsx

```diff
@@ -29,17 +36,130 @@ export type VolumeSnapshotModalProps = {
 export const VolumeSnapshotModal = withHandlePromise((props: VolumeSnapshotModalProps) => {
   const { close, cancel, pvcData, errorMessage, inProgress, handlePromise } = props;
   const resource = pvcData.data as K8sResourceKind;
+  const [snapshotCount, setSnapshotCount] = React.useState(3);
   const [snapshotName, setSnapshotName] = React.useState(
     `${getName(resource) || 'pvc'}-snapshot-1`,
   );
+  const [scheduleLabel, setScheduleLabel] = React.useState('');
   const snapshotTypes = {
-    single: 'Single Snapshot',
+    SingleSnapshot: 'SingleSnapshot',
+    CronJob: 'CronJob',
   };
+  const selectedKey = snapshotTypes.SingleSnapshot;
+  const [scheduleType, setScheduleType] = React.useState('');
 
+  const setSchedulerHandler = (value) => {
+    setScheduleType(snapshotTypes[value]);
+  };
+
+  const getCronElements = () => {
+    const numberOfSnapshot = {
+      '3': '3',
+      '5': '5',
+      '7': '7',
+    };
```

Comment:
IMO enum would be better. And maybe use that enum value to initialize the State as well. WDYT?